test/integration-tests/testexplorer/TestExplorerIntegration.test.ts

```diff
@@ -44,595 +46,596 @@ suite("Test Explorer Suite", function () {
     let workspaceContext: WorkspaceContext;
     let testExplorer: TestExplorer;
 
-    async function waitForTestExplorerReady(): Promise<vscode.TestController> {
-        return (
-            await Promise.all([
-                testExplorer.controller.items.size === 0
-                    ? eventPromise(testExplorer.onTestItemsDidChange)
-                    : Promise.resolve(testExplorer.controller),
-                syncPromise(() => vscode.commands.executeCommand("workbench.view.testing.focus")),
-            ])
-        )[0];
-    }
-
-    async function gatherTests(
-        controller: vscode.TestController,
-        ...tests: string[]
-    ): Promise<vscode.TestItem[]> {
-        const testItems = tests.map(test => {
-            const testItem = getTestItem(controller, test);
-            if (!testItem) {
-                const testsInController: string[] = [];
-                controller.items.forEach(item => {
-                    testsInController.push(
-                        `${item.id}: ${item.label} ${item.error ? `(error: ${item.error})` : ""}`
-                    );
-                });
+    suite("Debugging", function () {
+        let settingsTeardown: () => void;
 
-                assert.fail(
-                    `Unable to find ${test} in Test Controller. Items in test controller are: ${testsInController.join(", ")}`
-                );
-            }
-            assert.ok(testItem);
-            return testItem;
-        });
+        async function setup(settings: SettingsMap) {
```

Comment:
this would be a great common utility since we follow the same process everytime for a bunch of suites

Reply:
Moved to a common utility